benchto-driver/src/test/java/com/teradata/benchto/driver/execution/ExecutionDriverTest.java

```diff
@@ -112,6 +120,36 @@ public void benchmarkIsExecutedWhenNoTimeLimitEnds()
         verifyNoMoreInteractions(benchmarkExecutionDriver);
     }
 
+    @Test
+    public void failOnListenerFailure()
+    {
+        BenchmarkExecutionListener failingListener = new DefaultBenchmarkExecutionListener()
+        {
+            @Override
+            public Future<?> benchmarkFinished(BenchmarkExecutionResult benchmarkExecutionResult)
+            {
+                throw new IllegalStateException("programmatic listener failure in testFailingListener");
+            }
+        };
+
+        BenchmarkStatusReporter statusReporter = new BenchmarkStatusReporter(singletonList(failingListener));
+        /*
+         * Listeners are called by BenchmarkExecutionDriver so we need to provide one.
+         * Listeners results final check is invoked by ExecutionDriver, so this is tested here.
+         */
+        BenchmarkExecutionDriver benchmarkExecutionDriver = new BenchmarkExecutionDriver();
```

Comment:
can you please do not use `mock` or `ReflectionUtils`, can you just simply write this class?